src/lib/ffi/ffi_cipher.cpp

```diff
@@ -71,8 +78,9 @@ int botan_cipher_init(botan_cipher_t* cipher, const char* cipher_name, uint32_t
       }
 
       const size_t update_size = ffi_choose_update_size(*mode);
+      const size_t ideal_update_size = mode->ideal_granularity();
```

Comment:
This is potentially wrong if `ideal_granularity` is small and/or `minimum_final_size` is large. `ffi_choose_update_size` will scale up the granularity to satisfy the minimum final size. So we might end up here (in a weird situation) where the ideal update size was actually less than the update size.

I don't think this should happen absent an oversight in implementing a mode but should guard against it. Easiest path seems to be just use `std::max(mode->ideal_granularity(), update_size)`

Reply:
Fair enough. I applied your suggestion.